### ISSUE TITLE
Review feedback to address

### ISSUE THREAD

**Issue by gdalle** [adrhill/iclr2025-sparse-autodiff#59]
**Meta-reviewer**

- [ ] The most important one is that the blog post would be a lot stronger if it included an example from the machine learning literature that requires materialization of the Jacobian or Hessian matrix rather then access to these objects via vector products. In other words, please tell the ICLR audience why they should care about this post and take the time to read it. [...] Please give an example from machine learning that could take advantage of the machinery described in this blog post.
- [x] The less important one is the remark that the "...by people more familiar with low-level programming languages" comment is odd. I agree: better to simply say that this theory has been developed outside the ML community and leave it at that.
- [x] The caption of Figure 16 specifies the wrong f(x). Please fix this error.

**Reviewer 1**

- [x] ~~For example, reverse mode AD was invented as early as 1970, and before backprop was invented, even though the latter is a special case of the former. So discussing the history and connecting the two together, and explaining the difference, would be super interesting.~~
- [ ] Also, as the blog mentioned, in deep learning, typically there is only one loss function, so no Jacobi, so it would be nice to understand how the techniques in this paper could benefit the ML community. [...] ~~Adding the connection between the technique discussed here and ML would be super interesting. [...] History and connection with ML would make it stronger~~
- [x] "One reason is that the underlying theory was developed outside of the ML research ecosystem, by people more familiar with low-level programming languages." This seems a bit odd to say. ML people used to work in low-level programming languages as well. My understanding is that AD was invented for optimization by people who are numerical analysis researchers
- [x] " We consider a convolutional filter of size 5×5... an input size of 28x28x1...576×784 Jacobian" Would be helpful to explain where these numbers like 576 comes from.$

**Reviewer 2**

- [x] On a subjective note, a demonstration in Torch or JAX (both have 2nd order derivatives AFAIK) would have a greater impact over Julia, as more of the ML community uses these tools for real world applications and cases .
- [x] Consider replacing impossible with intractable when needed
- [x] Linear maps are formally defined (after figure 2), but a novice reader might miss their property, e.g. why are they merely an approximation for the Jacobian? -or- in what sense are the dashed matrices differ from the solid ones
- [x] Similar to the previous point, it is stated that materialization does not materialize any intermediate Jacobians, consider adding another sentence on when and especially why materialization differ.
- [x] Figure 6 - consider reminding the black-box here again - the numbers inside were not clear (to me) until a second read.
- [x] ~~Figure 13 - consider making this more general, each row has exactly two indexes :-)~~

**Comment by adrhill** [comment]
Adding this comment as a reminder to include our names and affiliations.

**Comment by gdalle** [comment]
Did you add our names and affiliations?

**Comment by adrhill** [comment]
See #62

**Comment by adrhill** [comment]
I'll let you decide whether this issue can be closed.

**Comment by gdalle** [comment]
I would probably move the applications section further up. And to justify the interest for Jacobians, recall that Newton's method also applies to more generic nonlinear systems of equations (which are not necessarily of the form $\nabla f(x) = 0$), in which case we solve a linear system involving the Jacobian.

**Comment by adrhill** [comment]
Sounds good to me!

**Comment by adrhill** [comment]
I'll leave it up to you, the build command that works for me is `rbenv exec bundle exec jekyll serve --future --open-url --livereload`.

**Comment by adrhill** [comment]
Keep in mind that we probably need to keep the section in a position where we already introduced our Matrix/Operator distinction.

**Comment by adrhill** [comment]
Sounds good to me. :)

**Comment by adrhill** [comment]
I'm assuming we're done?

**Comment by gdalle** [comment]
#63
